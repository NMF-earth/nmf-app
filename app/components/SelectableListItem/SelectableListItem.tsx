import React from "react";
import { TouchableOpacity } from "react-native";
import { Ionicons } from "@expo/vector-icons";

import styles from "./SelectableListItem.styles";
import Text from "../../components/Text";
import { Colors } from "../../style";

interface Props {
  selected: boolean;
  title: string;
  onPress: () => void;
}

const SelectableListItem = ({ selected, title, onPress }: Props) => {
  return (
    <TouchableOpacity onPress={onPress} style={styles.container}>
      <Text.Primary bold numberOfLines={1} green={selected} style={styles.text}>
        {title}
      </Text.Primary>
      {selected && (
        <Ionicons
          name={"md-checkmark"}
          size={26}
          style={styles.icon}
          color={Colors.green50}
        />
      )}
    </TouchableOpacity>
  );
};

export default SelectableListItem;
