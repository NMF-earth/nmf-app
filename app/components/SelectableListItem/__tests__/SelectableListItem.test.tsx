import "react-native";
import React from "react";
import { create } from "react-test-renderer";

import SelectableListItem from "../SelectableListItem";

describe("<SelectableListItem />", () => {
  const defaultProps = {
    selected: true,
    title: "France",
    onPress: () => {
      // do nothing.
    },
  };
  const wrapper = create(<SelectableListItem {...defaultProps} />);

  test("render", () => {
    expect(wrapper).toMatchSnapshot();
  });

  const wrapperNotSelected = create(
    <SelectableListItem selected={false} {...defaultProps} />
  );

  test("render if not selected", () => {
    expect(wrapperNotSelected).toMatchSnapshot();
  });
});
