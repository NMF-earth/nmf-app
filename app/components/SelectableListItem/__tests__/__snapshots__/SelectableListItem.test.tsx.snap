// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectableListItem /> render 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    {
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <Text.Primary
    bold={true}
    green={true}
    numberOfLines={1}
    style={
      {
        "paddingVertical": 20,
      }
    }
  >
    France
  </Text.Primary>
  <Ionicons
    color="#0F853B"
    name="md-checkmark"
    size={26}
    style={
      {
        "marginTop": 18,
      }
    }
  />
</TouchableOpacity>
`;

exports[`<SelectableListItem /> render if not selected 1`] = `
<TouchableOpacity
  onPress={[Function]}
  style={
    {
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <Text.Primary
    bold={true}
    green={true}
    numberOfLines={1}
    style={
      {
        "paddingVertical": 20,
      }
    }
  >
    France
  </Text.Primary>
  <Ionicons
    color="#0F853B"
    name="md-checkmark"
    size={26}
    style={
      {
        "marginTop": 18,
      }
    }
  />
</TouchableOpacity>
`;
